app/controllers/publishers_controller.rb

```diff
@@ -422,13 +422,23 @@ def balance
     end
   end
 
+  def get_site_banner_data
+    default_site_banner_mode = current_publisher.default_site_banner_mode
+    default_site_banner = {"id" => current_publisher.default_site_banner_id, "name" => "Default", "type" => "Default"}
+    channel_banners = current_publisher.channels.map { |channel| {id: channel.id, name: channel.publication_title, type: channel.details_type} }
+    channel_banners.each do |channel_banner|
+      channel_banner[:id] = current_publisher.site_banners.find_by(channel_id: channel_banner[:id]).id
```

Comment:
I'm a little bit confused about why you're replacing the id of each channel banner with the id of the site banner?
I think in your map above it would be easier to do this
```ruby
channel_banners = current_publisher.channels.map do |channel|
  {
    id: channel.id,
    site_banner_id: channel.site_banner.id,
    name: channel.publication_title,
    type: channel.details_type
  }
end
```

Reply:
Thanks for the tip on using channel.site_banner.id 😎